charforge-gui/backend/app/services/charforge_integration.py

```
@@ -0,0 +1,285 @@
+        # This is a simplified version - you might need to adjust based on actual output
+        if "Step" in line and "/" in line:
+            try:
+                # Extract step numbers like "Step 100/800"
+                parts = line.split()
+                for i, part in enumerate(parts):
+                    if part == "Step" and i + 1 < len(parts):
+                        step_info = parts[i + 1]
+                        if "/" in step_info:
+                            current, total = step_info.split("/")
+                            return float(current) / float(total) * 100
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)